### PR TITLE
fixes #5197 Preallocated disk support for ovirt 

### PR DESCRIPTION
**Pull request by leon-strong**
Please excuse my github newbieness, this should replace PR #1568

### CODE REVIEW

**Comment by mmoll** [comment]
You can modify existing pull requests by adding more commits to your branch. Once the review is done, you will normally be asked to "squash" your commits. This means, you use git rebase -i to melt all your commits into one and then force push your new branch. For now, however, run the [test]

**Comment by lzap** [comment]
Ok that looks better.

Screenshot

I tested this and it works fine. Maybe instead _Thick Provision the disk when creating_ I'd go for _Uncheck for Thin Provision option_

I tested this and it works for both Preallocated and Thin Provision against RHEV 3.3. Please squash into one commit.

File: app/views/compute_resources_vms/form/ovirt/_volume.html.erb [hunk 1]
```diff
@@ -9,6 +9,9 @@
                  :help_inline=> remove_child_link("X", f, { :method => :'_delete', :title => _('remove volume'), :class => 'label label-danger' }) %>
     <%= f.hidden_field :storage_domain if disabled %>
     <%= f.hidden_field :id %>
+    
+    <%= checkbox_f f, :preallocate, { :checked => true, :help_inline => _("Thick Provision the disk when creating"), :label => _('Preallocate Disk') } %>
```
Comment (domcleal) [inline app/views/compute_resources_vms/form/ovirt/_volume.html.erb]: Please also change the first letter all "Provision" and "Disk" to lower case, only the first should be upper case.

**Comment by leon-strong** [comment]
I kinda like "Uses thin provisioning if unchecked" as suggested on IRC, any other suggestions?

**Comment by lzap** [comment]
Hmmm we use this format for the commit messages:

```
fixes #5197 - Preallocated disk support for ovirt
```

We are good to go, Dominic will likely do it in the next merge timeframe :-) Thanks for your contribution.

**Comment by domcleal** [comment]
[test]  If you're happy, you can merge it @lzap.  I still disagree somewhat with the default due to the inconsistency with other CRs, but otherwise fine.

**Comment by lzap** [comment]
> [test]  If you're happy, you can merge it @lzap.  I still disagree somewhat with the default due to the inconsistency with other CRs, but otherwise fine.

While thin provisioning is not a bad concept at all, it allows
over-provisioning which can (when incorrectly approached by an
administrator) lead to data loss.

Having said that, modern VM managers warn quite early to prevent from
that situation, including RHEV. I am fine with having thin option as the
default one.

Others have opinions about that?

## 

Later,

 Lukas "lzap" Zapletal
 irc: lzap #theforeman

**Comment by domcleal** [comment]
I've no opinion about thin provisioning itself, btw, but my point is purely one of consistency between Foreman's CRs.  I happen to use thin provisioning with libvirt, but if it were disabled by default then I'd enable it via a profile.

**Comment by leon-strong** [comment]
I've been doing some hard thinking last night, and I think dom's right for a few reasons, firstly because we're now changing an existing default behaviour, not just that, we can't be sure that the additional provisioning time that'd be introduced because of that default change won't adversely affect existing installations. (causing build timeouts etc is a real possibility). 

I think the prudent thing here is to simply add the support, keep existing default behaviour, and be comfortable in the fact if users wish to change their default settings they can do it via profiles/compute templates, and we can be sure that this change won't adversely affect existing users

**Comment by domcleal** [comment]
Thanks for revisiting that @icesmurf, much appreciated!  [test]

**Comment by lzap** [comment]
I am fine with that. Nice contribution, thanks a lot!

## 

Later,

 Lukas "lzap" Zapletal

**Comment by domcleal** [comment]
Merged as 2bf991a for Foreman 1.6.0.  Thanks for your contribution @icesmurf!

**Pull request closed without merging**